src/ws/update-motd

```diff
@@ -12,13 +12,25 @@ port=${port:-9090}
 # hostname from cmdline, then `hostname -f`
 hostname=${2:-$(hostname -f)}
 
+# determine the first interface with a default route
+interface6=${3:-$(ip -o -6 route get :: 2>/dev/null | sed -e 's/.*dev \([^ ]*\) .*/\1/')}
+# use the interface to determine the global ip6 address
+if [ ! -z $interface6 ]; then
+    ip6=${3:-$(ip -o -6 addr show dev $interface6 scope global | sed -e 's/.*inet6 \([^ ]*\) .*/\1/' | sed -e 's/\/.*//')}
+fi
+
 # ip addr from cmdline, then default route source addr
-ip=${3:-$(ip -o route get 255.0 2>/dev/null | sed -e 's/.*src \([^ ]*\) .*/\1/')}
+ip4=${3:-$(ip -o route get 255.0 2>/dev/null | sed -e 's/.*src \([^ ]*\) .*/\1/')}
 
 # protocol from cmdline, then https
 protocol=${4:-https}
 
 hostname_url="${protocol}://${hostname}:${port}/"
-ip_url="${ip:+ or ${protocol}://${ip}:${port}/}"
+# if a ip6 global address is present use it for ip_url otherwise display the ip4 address
```

Comment:
Can we please flip this around? If there is an IPv4 address, it's much simpler to handle than an IPv6 one. Particularly on a VT where you don't have copy&paste, IPv6 addresses are rather unwieldy. (In general we want to present host names when possible, of course.)

Reply:
@martinpitt I agree hostname is the easiest and preferred method by far! I displayed the IPv6 address over the IPv4 address only when a global unicast address is present. This would mean either intentionally or not the system likely has a functional IPv6 stack and it's being preferred by the underlying operating system. This is the larger trend in the networking world. Additionally most organizations are assigned stable IPv6 prefixes. These will and are memorized by IT staff just like IPv4 address space is memorized today. This cuts down on the number of bytes the user has to remember on that sprint back to a GUI enabled terminal.

The focus of your point is on usability. In that regard I'd be a fool to say an IPv6 address is easier to remember by a glance at the console without copy and paste. That said, by displaying the IPv4 address over the IPv6 we'd be reversing the position of the rest of the network stack and the RFC world.

My preference would be to stay consistent and display useful IPv6 information when it's available by default.

Alternatively - we could display both IPv4 and IPv6 URLs.

Lastly - what's easier to remember: https://127.0.0.1:9090 or https://[::1]:9090 so not all IPv6 addresses are bad right?